### PR TITLE
Gradle and build scripts updates

### PR DESCRIPTION
**Pull request by Tapchicoma**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,8 +1,3 @@
...
-kotlin_version=2.0.0-RC3
-coroutines_version=1.8.0
-logback_version=1.3.14
-datetime_version=0.6.0-RC.2
-ktor_version=2.3.10
-retrofit_version=2.11.0
-serialization_version=1.6.3
\ No newline at end of file
+
+org.gradle.configuration-cache=true
```
Comment (adam-enko) [inline gradle.properties:3]: WDYT about enabling Build Cache and Parallel too?

```suggestion
org.gradle.configuration-cache=true
org.gradle.caching=true
org.gradle.parallel=true
```

**Comment by Tapchicoma** [comment]
Last commit introduced workaround to manage running server in background. This server is  automatically started on running any app configuration. If Gradle daemon is stopped - server is stopped as well. It is also possible to control server either via modified IDEA run configurations or via `startServerInBackground` and `stopServerInBackground` tasks.

While I am confident that  it will work for single Gradle daemon case (server in background + app)  - in your workshop you advice at some point to start the second chat app which starts second Gradle daemon instance. In my tests such scenario works, but my confidence here is around 80%. If you want  to proceed with  existing approach - I will remove last commit from this PR.

**Comment by SebastianAigner** [comment]
@Tapchicoma For me personally, 80% is a little low, and I'd like to avoid any Gradle-run related issues as much as possible 😅

**Comment by Tapchicoma** [comment]
@SebastianAigner I've removed the last commit. 

Please note that Gradle daemon still should run on JDK 11+ because of the compose compiler plugin, which was compiled against JDK 11 in Kotlin `2.0.0` release. Compose plugin should be compatible with JDK 1.8 in the next Kotlin release `2.0.20`.

Produced exception suggests several ways to change JDK for Gradle wrapper.

**Comment by SebastianAigner** [comment]
Thank you! Cherry picked this into the `solutions-new` branch and rebased master on top of it! Really appreciate the Gradle contribution, Yahor! ❤️

**Pull request closed without merging**